packages/main/src/FileUploader.js

```diff
@@ -0,0 +1,320 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import ValueState from "@ui5/webcomponents-base/dist/types/ValueState.js";
+import { getFeature } from "@ui5/webcomponents-base/dist/FeaturesRegistry.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import {
+	fetchI18nBundle,
+	getI18nBundle,
+} from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import {
+	FILE_UPLOADER_BROWSE_TEXT,
+} from "./generated/i18n/i18n-defaults.js";
+import Button from "./Button.js";
+import Input from "./Input.js";
+
+// Template
+import FileUploaderTemplate from "./generated/templates/FileUploaderTemplate.lit.js";
+
+// Styles
+import FileUploaderCss from "./generated/themes/FileUploader.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-file-uploader",
+	properties: /** @lends sap.ui.webcomponents.main.FileUploader.prototype */ {
+		/**
+		 * Comma-separated list of file types that the <code>ui5-file-uploader</code> should accept.
+		 * @type {string}
+		 * @public
+		 */
+		accept: {
+			type: String,
+		},
+
+		/**
+		 * If set to "true", the <code>ui5-file-uploader</code> will be rendered as button only, without showing the input field.
+		 * @type {boolean}
+		 * @public
+		 */
+		buttonOnly: {
+			type: Boolean,
+		},
+
+		/**
+		 * Defines whether <code>ui5-file-uploader</code> is in disabled state.
+		 * <br><br>
+		 * <b>Note:</b> A disabled <code>ui5-input</code> is completely noninteractive.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		disabled: {
+			type: Boolean,
+		},
+
+		/**
+		 * Defines the icon to be displayed as graphical element within the button.
+		 * See all the available icons in the <ui5-link target="_blank" href="https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html" class="api-table-content-cell-link">Icon Explorer</ui5-link>.
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		icon: {
+			type: String,
+		},
+
+		/**
+		 * If set to true, the button will show icon at the end.
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		iconEnd: {
+			type: Boolean,
+		},
+
+		/**
+		 * If set to true, the button is displayed without any text.
+		 * @type {boolean}
```

Comment:
In my opinion, instead of icon, iconOnly, iconEnd we can think of a slot and whatever element the user pass in that slot, that element would be the file browse trigger.